frontend/src/containers/RequiresEdit.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { withProfile } from "../providers/ProfileProvider";
+
+/**
+ * A component which renders its children only if the user has editable permission
+ * in their current context.
+ */
+const RequiresEdit = (
+  { profile, channel, displayOnFalse, component: Component, children, ...otherProps }
+) => {
+  let display = '';
+
+  if (profile && profile.channels) {
+    display = displayOnFalse;
+
+    if (channel) {
+      // if a channel is given, check that the user can edit it.
+      if (profile.channels.find((el) => (el.id === channel.id))) {
+        display = children;
+      }
+    } else {
+      // if no channel is given, check the user has channels they can edit.
+      if (profile.channels.length > 0) {
+        display = children;
+      }
```

Comment:
Since this component does two things, I think it should be two components: HiddenIfNoChannels and HiddenIfNotOwnChannel. The HiddenIfNoChannels component checks if there are any channels in the profile at all and the HiddenIfNotOwnChannel component checks if a specific channel is in the profile. They are two related, but different things. Finally, the components are checking if a profile has channels so that should be explicit in the names.